### DISCUSSION TITLE
Peicewise linear encoding location?

### DISCUSSION THREAD

**Discussion by pqrz** [yandex-research/rtdl-num-embeddings#1] [Q&A] [answered]
Great work!
I am unable to locate piecewise linear encoding in this code - rtdl. 
If possible, could you point me to it?

**Comment by Yura52** [comment, marked as answer]
> Great work!

Thanks! 

> I am unable to locate piecewise linear encoding in this code

Thanks for the question! We need to add more instructions to README in that regard.

- paper implementation: https://github.com/Yura52/tabular-dl-num-embeddings
- all neural networks are implemented in `bin/train4.py`. For PLE-based models, you should use the [`bins` option](https://github.com/Yura52/tabular-dl-num-embeddings/blob/6758410e97d0a7d7a521715447cf1c425350520a/bin/train4.py#L43)
- then, check the configuration files in the `exp` folder. The inner folder names follow the names from the paper.  For example, here is the directory for tuning and evaluation of MLP-T-LR on the California Housing dataset: [link](https://github.com/Yura52/tabular-dl-num-embeddings/tree/main/exp/mlp-t-lr/california).
- note that in this specific case, the older version of train4.py, namely train1.py, was used, which can be found by looking at the [tuning config](https://github.com/Yura52/tabular-dl-num-embeddings/blob/6758410e97d0a7d7a521715447cf1c425350520a/exp/mlp-t-lr/california/0_tuning.toml#L2)
- so, to run the already tuned MLP-T-LR on the California Housing dataset, you can copy any config from the evaluation directory and run train1.py with it
    1. `cp exp/mlp-t-lr/california/0_evaluation/0.toml exp/mlp-t-lr/california/0_evaluation/0_reproduce.toml`
    2. `python bin/train1.py exp/mlp-t-lr/california/0_evaluation/0_reproduce.toml`
- however, for futher usage, we recommend adapting the configs for `bin/train4.py`

Does this help?

**Comment by Yura52** [comment]
P.S. as of now, in the `rtdl` library, we do not provide the embedding modules from the paper

**Comment by Yura52** [comment]
Hi @pqrz ! Just in case, sharing that I created an easy-to-use package for this paper "On Embeddings for Numerical Features in Tabular Deep Learning":
- [The package](https://github.com/yandex-research/rtdl-num-embeddings/tree/main/package#python-package-)
- [The announcement](https://twitter.com/YuraFiveTwo/status/1737478716685447606) with more details

P.S. More projects on tabular deep learning are available [here](https://github.com/yandex-research/rtdl).
